node/pkg/fetcher/collector.go

```diff
@@ -42,43 +42,84 @@ func (c *Collector) Run(ctx context.Context) {
 }
 
 func (c *Collector) Job(ctx context.Context) error {
-	log.Debug().Str("Player", "Collector").Str("collector", c.Name).Msg("collectorJob")
-	rawResult, err := c.collect(ctx)
+	feeds, err := c.collect(ctx)
 	if err != nil {
-		log.Error().Str("Player", "Collector").Err(err).Msg("error in collect")
 		return err
 	}
 
-	if len(rawResult) == 0 {
+	if len(feeds) == 0 {
 		return nil
 	}
 
-	aggregated, err := calculator.GetFloatMed(rawResult)
+	return c.processFeeds(ctx, feeds)
+}
+
+func (c *Collector) processFeeds(ctx context.Context, feeds []FeedData) error {
+	if isFXPricePair(c.Name) {
+		return c.processFXPricePair(ctx, feeds)
+	}
+	return c.processVolumeWeightedFeeds(ctx, feeds)
+}
+
+func (c *Collector) processFXPricePair(ctx context.Context, feeds []FeedData) error {
+	median, err := calculateMedian(feeds)
+	if err != nil {
+		return err
+	}
+	return insertAggregateData(ctx, c.ID, median)
+}
+
+func (c *Collector) processVolumeWeightedFeeds(ctx context.Context, feeds []FeedData) error {
+	volumeWeightedFeeds := filterFeedsWithVolume(feeds)
+	vwap, err := calculateVWAP(volumeWeightedFeeds)
 	if err != nil {
-		log.Error().Str("Player", "Collector").Err(err).Msg("error in GetFloatMed")
 		return err
 	}
-	err = insertLocalAggregateRdb(ctx, c.ID, aggregated)
+	median, err := calculateMedian(feeds)
```

Comment:
@nick-bisonai I think the second component (median) of the aggregated price should be computed only from values that have not been used for volume computation 🤔

Reply:
I thought this would be safer, but if you insist, I'll update that way